### DISCUSSION TITLE
The role of the setMatcher method in the SearchFiled

### DISCUSSION THREAD

**Discussion by robbiezl** [dlsc-software-consulting-gmbh/GemsFX#176] [General]
I have reviewed and run the provided SearchFieldApp example, but still cannot fully understand the roles of setSuggestionProvider and setMatcher. During the runtime, I tried to comment out the differences between viewing and running separately, but still couldn't understand the differences between the two, especially the setMatcher method.

May I ask if   there is a more colloquial way to explain it.

**Comment by dlemmermann** [comment]
The suggestion provider is responsible for returning a collection of all possible values that might be useful for the user based on the current text input. The matcher is then used to determine which one of those suggestions is the best match. The best match is usually the item that starts with exactly the text entered so far. See also Javadocs:

```java
    /**
     * The function that is used to determine the first item in the suggestion list that is a good match
     * for auto selection. This is normally the case if the text provided by the converter for an item starts
     * with exactly the text typed by the user. Auto selection will cause the field to automatically complete
     * the text typed by the user with the name of the match.
     *
     * @return the function used for determining the best match in the suggestion list
     * @see #converterProperty()
     */
```

**Comment by dlemmermann** [comment]
@robbiezl Please let us know if you think that there is a semantic mismatch somewhere or something simply not being intuitive enough or maybe even redundant.

**Comment by dlemmermann** [comment]
We also have a WIKI page for the SearchField https://github.com/dlsc-software-consulting-gmbh/GemsFX/wiki/SearchField

**Reply by robbiezl** [reply]
Thank you for your patient reply. The wiki was written in great detail
